core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -46,7 +46,7 @@
     super(ops, table, name);
 
     this.schema = new Schema(
-        Types.NestedField.required(1, "partition", table.spec().partitionType()),
+        Types.NestedField.required(1, "partition", table.spec().maxPartitionType(table.specs())),
         Types.NestedField.required(2, "record_count", Types.LongType.get()),
         Types.NestedField.required(3, "file_count", Types.IntegerType.get())
     );
```

Comment:
use max to solve after spark WholeStageCodegen, getType error. 

If not, when use `days(ts)`, then drop it. The spark will use ts type: timestamp in schema, then genCode is `project_tmpInput_0.getLong(0)`. But the partition data value type is int.